### ISSUE TITLE
Forward declare of enum class in template argument is a Microsoft extension

### ISSUE THREAD

**Issue by kokole** [KN4CK3R/UnrealEngineSDKGenerator#11]
Hello, currently the generated SDK can only be compiled with MSVC because the enum members of the generated classes (like `TEnumAsByte<enum class ENetRole>`) aren't standard C++.

I propose 3 solutions:
- Fix forward declaration somehow?
- Put the enums in x_structs.hpp after constants and before structs (don't know how that would work with package ordering though).
- Change `TEnumAsByte<enum class ESomeEnum>` to `uint8_t/*ESomeEnum*/` or `TEnumAsByte<uint8_t/*ESomeEnum*/>`

**Comment by KN4CK3R** [comment]
Didn't know only MSVC can do this. Could you test if the SDK works after changing to code generation in GenericTypes.cpp to
```
return Info::Create(PropertyType::Primitive, sizeof(uint8_t), false, "TEnumAsByte<" + MakeUniqueCppName(enumObj) + ">");
```
The enums should be known at every class, so the forward declaration may not be necessary. If that doesn't work, the "put them into an other file" could be a good solution.

**Comment by kokole** [comment]
No, that doesn't work. It works better if combined with the "enums in another x_structs.hpp", but I get a name collision in PUBG (between a member variable and an enum).
What is wrong with the other compiler (Clang) is that the compiler shouldn't know the size of an enum, so it fails. I tried adding " : uint8_t" in various forms (lots of them) but still didn't make it work.
So I used the uint8_t/* ESomeEnum */ solution.
